### DISCUSSION TITLE
Add an option to auto download a beatmap after pressing osu!direct on the website

### DISCUSSION THREAD

**Discussion by crumpledmint** [ppy/osu#30638] [Ideas]
So basically stable auto downloads any beatmap that you click "osu!direct" on and don't have downloaded at the moment but with lazer it opens up a beatmap page in the lazer's browser instead of downloading it. This drastically increases time to download multiple maps when you browse ranked/search section and m3 multiple beatmaps and then bulk download it by pressing osu!direct while tabbing between all of the opened beatmap pages because you need to manually confirm every download by alt tabbing into lazer every time. I suggest making it auto download instead and also in case of you already having a beatmap auto redownload it because there are literally 0 cases when you browse osu web and click "osu!direct" without intention to download the beatmap.

**Comment by BloodLantern** [comment]
I agree with this.
Now every time I click on the osu!direct button, I hesitate to instead download the beatmap manually as it would be faster, but that's literally the opposite of what osu!direct is meant to be: a faster and more convenient way to download beatmaps.

**Comment by WalmartWhiteCat** [comment]
So like, why are you guys using the osu! website to download maps when you play Lazer? Would it not be easier to just use the Lazer map listing to find what you want and then download everything _in_ Lazer instead of through the website? I mean, I can see a valid argument for this when it comes to downloading premade packs. But for downloading maps individually, using the built-in map listing for Lazer is already more convenient.

I think adding the ability to download packs natively on the Lazer client should be focussed on if it’s a real issue for people, rather than adding ultimately redundant functions to an already streamlined process (like downloading individual maps).

**Reply by BloodLantern** [reply]
I agree that most of the time, using the in-game beatmap listing is fine. But when it comes to opening a map link from outside the game, like from Discord for example, it opens up the website and therefore would be much more convenient if the osu!direct button allowed to automatically download the beatmap in addition to opening its page in-game. Or maybe make it a setting that people can change, like with the option to either open the beatmap page, download the beatmap, or do both.

Anyway, the more Lazer gets improved, the less we will need to use the website, so you're right when saying that we should use the in-game beatmap listing.
It's just that for now it misses some convenience features, such as when you're on someone's profile and want to check one of the maps they played. If you go on a beatmap's page from a player profile, when trying to go back, you need to reopen the profile page. This means you can't easily download multiple maps from someone's profile, hence using the website for such a thing instead.

Edit: The issue I'm talking about already exists here: #5063

**Reply by crumpledmint** [reply]
I do that but honestly the download button is so small in the beatmaps section that it's faster for me to open the map page and download it from there. When you are downloading one map it's fine but when you download like 10 at a time it's unbearably slower than just pressing o!direct ctrl+w multiple times

**Comment by shiumano** [comment]
It is honestly a bit silly to see the beatmap page once on the web and then again on the lazer.
Perhaps there should be a toggle in the “online” section of the settings for “osu!direct link to direct download/jump to map”.

**Comment by ziv-vy** [comment]


**Reply by frenzibyte** [reply]
I'm failing to see how this ties to anything else in the discussion thread.
